mbed/mbed.py

```diff
@@ -2685,6 +2689,54 @@ def test_(toolchain=None, target=None, compile_list=False, run_list=False, compi
     program.set_defaults(target=target, toolchain=tchain)
 
 
+# device management commands
+@subcommand('device-management',
+    dict(name=['-t', '--toolchain'], help='Toolchain used for mbed compile'),
+    dict(name=['-m', '--target'], help='Target used for compile for target MCU. Example: K64F, NUCLEO_F401RE, NRF51822...'),
+    dict(name=['--profile'], help=""),
+    dict(name='--build', help='Build directory. Default: build/'),
+    dict(name='--source', action='append', help='Source directory. Default: . (current dir)'),
+    help='device management supcommand',
+    hidden_aliases=['dev-mgmt', 'dm'],
+    description=("Manage Device with Palion"))
+def dev_mgmt(toolchain=None, target=None, source=False, profile=False, build=False):
+    orig_path = getcwd()
+    program = Program(getcwd(), True)
+    program.check_requirements(True)
+    with cd(program.path):
+        tools_dir = program.get_tools()
+
+    script = os.path.join(tools_dir, 'device_management.py')
+    if not os.path.exists(script):
+        error('device management is not supported by this version of Mbed OS. Please upgrade.')
+
+
+    target = target or program.get_cfg("TARGET")
+    toolchain = toolchain or program.get_cfg("TOOLCHAIN")
+
+    if build:
+        build_path = build
+    elif not build and target and toolchain:
```

Comment:
I'll nit. I understand this as (not A) and (B) and (C), but might be misunderstood as not (A and B and C).
Might make it easier to understand if parentheses were explicit.